src/ui/popup.js

```diff
@@ -283,18 +322,18 @@ export default class Popup extends Evented {
         }
 
         this._content = DOM.create('div', 'mapboxgl-popup-content', this._container);
-
         if (this.options.closeButton) {
             this._closeButton = DOM.create('button', 'mapboxgl-popup-close-button', this._content);
             this._closeButton.type = 'button';
             this._closeButton.setAttribute('aria-label', 'Close popup');
             this._closeButton.innerHTML = '&#215;';
             this._closeButton.addEventListener('click', this._onClickClose);
         }
+
     }
 
-    _update() {
-        if (!this._map || !this._lngLat || !this._content) { return; }
+    _update(cursor: PointLike) {
+        if (!this._map || !this._lngLat && !this._trackPointer || !this._content) { return; }
```

Comment:
for clarity's sake, you may want to explicitly group the middle term here: `!this._map || (!this._lngLat && !this._trackPointer) || !this._content`